File: Baekjoon_Online_Judge/NO_18868/18868.cpp
#include <iostream>
using namespace std;
/*
 -------------------------------------- 
 | Created by happiness96             | 
 | Year 2021                          | 
 | Month 06                           | 
 | Day 27                             | 
 | 18868 멀티버스 Ⅰ                   | 
 -------------------------------------- 
*/

int main(){
    int M, N, result = 0;

    cin >> M >> N;

    int planets[M][N];

    for(int i = 0; i < M; i ++){
        for(int j = 0; j < N; j ++){
            cin >> planets[i][j];
        }
    }

    for(int i = 0; i < M; i ++){
        for(int j = i + 1; j < M; j ++){
            int flag = 1;

            for(int a = 0; a < N; a ++){
                for(int b = 0; b < N; b ++){
                    if(planets[i][a] == planets[i][b] and planets[j][a] == planets[j][b]) continue;
                    else if(planets[i][a] > planets[i][b] and planets[j][a] > planets[j][b]) continue;
                    else if(planets[i][a] < planets[i][b] and planets[j][a] < planets[j][b]) continue;
                    flag = 0;
                    break;
                }
                if(flag == 0) break;
            }
            result += flag;
        }
    }

    cout << result;
    return 0;
}

File: Baekjoon_Online_Judge/NO_18868/18868.py
# -*- encoding: utf-8 -*-
import sys
r_input = sys.stdin.readline

# -------------------------------------- #
# | Created by happiness96             | #
# | Year 2021                          | #
# | Month 07                           | #
# | Day 03                             | #
# | 18868 멀티버스 Ⅰ                   | #
# -------------------------------------- #


def run():
    M, N = map(int, r_input().split())

    planets = [list(map(int, r_input().split())) for _ in range(M)]

    result = 0

    for A in range(M):
        for B in range(A + 1, M):
            flag = 1

            for i in range(N):
                for j in range(N):
                    if planets[A][i] > planets[A][j] and planets[B][i] > planets[B][j] or planets[A][i] == planets[A][j] and planets[B][i] == planets[B][j] or planets[A][i] < planets[A][j] and planets[B][i] < planets[B][j]:
                        continue
                    flag = 0
                    break
                
                if not flag:
                    break
            
            result += flag
    
    print(result)


if __name__ == "__main__":
    run()
